sentry-ruby/lib/sentry/log_event_buffer.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+require "sentry/threaded_periodic_worker"
+
+module Sentry
+  class LogEventBuffer < ThreadedPeriodicWorker
+    FLUSH_INTERVAL = 5 # seconds
+    DEFAULT_MAX_EVENTS = 100
+
+    def initialize(configuration, client)
+      super(configuration.logger, FLUSH_INTERVAL)
+      @client = client
+      @pending_events = []
+      @max_events = configuration.max_log_events || DEFAULT_MAX_EVENTS
+      @dsn = configuration.dsn
+      @sdk = Sentry.sdk_meta
+      @mutex = Mutex.new
+
+      log_debug("[LogEvents] Initialized buffer with max_events=#{@max_events}, flush_interval=#{FLUSH_INTERVAL}s")
+    end
+
+    def start
+      ensure_thread
+      self
+    end
+
+    def flush
+      @mutex.synchronize do
+        return unless size >= @max_events
```

Comment:
in the case where someone adds only say 5 logs, these will never get sent, even at the exit flush

Reply:
what you want instead is 
* a periodic flush all the time
* a force flush when the size exceeds while adding